source/fab/steps/compile_c.py

```diff
@@ -30,132 +31,134 @@
 DEFAULT_OUTPUT_ARTEFACT = ''
 
 
-class CompileC(Step):
+@dataclass
+class MpCommonArgs(object):
```

Comment:
This pattern now appears in steps which call run_mp. We used to just bung some protected variables in the step class, for the child processes to read. Now we bundle it all up into a data class and pass it in with the list of items.

Reply:
Not as a context, it's additional data passed in to run_mp.
Instead of `run_mp([file1, file2, file3])`
we now have `run_mp([(file1, args), (file2, args), (file3, args)])`.